packages/core/src/queue.js

```diff
@@ -27,7 +27,9 @@ export default class Queue {
 
   // Resolves when there are no more queued or pending tasks.
   idle() {
-    return idle(() => this.length);
+    return waitFor(() => this.length === 0, {
+      timeout: 2 * 60 * 1000 // 2 minutes
```

Comment:
How did you settle on 2 mins?

Reply:
It was indefinite before and would just hang if a queued task never resolves for any reason. Figured 2 minutes was long enough to finish uploading any snapshots + resources that weren't uploaded during the rest of the run. We can adjust as needed, but this is really just a fallback in case a queued task hangs somewhere.